api/plugins/action/deploy_target_config.py

```diff
@@ -24,44 +17,67 @@ def run(self, tmp=None, task_vars=None):
 
         self.createClient(task_vars)
 
-        display.vvv("With: " + self._task.args.get('project'))
-        lagoonProject = Project(self.client).byName(self._task.args.get('project')).withDeployTargetConfigs()
+        self._display.vvv(f"With: {self._task.args}")
+        lagoonProject = Project(self.client).byName(
+            self._task.args.get('project'),
+            ['id', 'name']).withDeployTargetConfigs()
```

Comment:
Specifying just these two fields significantly decreases the query length and therefore reduces the number of authorisation checks that the Lagoon API must do for each field.